src/compiler/scala/tools/nsc/backend/jvm/BCodeHelpers.scala

```diff
@@ -631,7 +631,7 @@ abstract class BCodeHelpers extends BCodeIdiomatic {
         var access = asm.Opcodes.ACC_FINAL
         if (param.isArtifact)
           access |= asm.Opcodes.ACC_SYNTHETIC
-        jmethod.visitParameter(param.name.decoded, access)
+        jmethod.visitParameter(param.name.encoded, access)
```

Comment:
👍

Scala's name encoding keeps `isJavaIdentifierPart` characters, so there should not be any drawback in Java interop for legal Java identifiers.